### PR TITLE
Creating csv, crd, and packages for clusterlogging and elasticsearch …

### PR DESCRIPTION
**Pull request by jcantrill**
…operators

This PR supercedes https://github.com/operator-framework/operator-lifecycle-manager/pull/555  and includes the code review items

### CODE REVIEW

File: deploy/chart/catalog_resources/rh-operators/elasticsearch.v0.0.1.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        path: configMapName
+        x-descriptors:
+          - 'urn:alm:descriptor:io.kubernetes:ConfigMap'
+      - description: The name of the secret used by the Elasticsearch pods
+        displayName: Secret
+        path: secretName
+        x-descriptors:
+          - 'urn:alm:descriptor:io.kubernetes:Secret'
+      - description: Limits describes the minimum/maximum amount of compute resources required/allowed
+          displayName: Resource Requirements
```
Comment (alecmerdler) [inline deploy/chart/catalog_resources/rh-operators/elasticsearch.v0.0.1.clusterserviceversion.yaml]: Fix indentation here.

File: deploy/chart/catalog_resources/rh-operators/clusterloggings.crd.yaml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                              description: 'Limits describes the maximum amount of compute
+                                resources allowed. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/'
+                              type: object
+                            requests:
+                              description: 'Requests describes the minimum amount of compute
+                                resources required. If Requests is omitted for a container,
+                                it defaults to Limits if that is explicitly specified, otherwise
+                                to an implementation-defined value. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/'
+                              type: object
+                    required:
```
Comment (alecmerdler) [inline deploy/chart/catalog_resources/rh-operators/clusterloggings.crd.yaml]: De-indent this by one level (needs to be same level as `kibana.properties`).

**Comment by ecordell** [comment]
You can run `make schema-check` locally, which should help you find some of the formatting issues early

File: deploy/chart/catalog_resources/rh-operators/clusterlogging.v0.0.1.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        displayName: Kibana Size
+        path: visualization.kibana.replicas
+        x-descriptors:
+        - 'urn:alm:descriptor:com.tectonic.ui:podCount'
+      - description: Resource requirements for the Kibana pods
+        displayName: Kibana Resource Requirements
+        path: visualization.kibana.resources
+        x-descriptors:
+        - 'urn:alm:descriptor:com.tectonic.ui:resourceRequirements'
+      - descriptor: The node selector to use for the Kibana Visualization component
```
Comment (alecmerdler) [inline deploy/chart/catalog_resources/rh-operators/clusterlogging.v0.0.1.clusterserviceversion.yaml]: This needs to be `description`.

File: deploy/chart/catalog_resources/rh-operators/clusterlogging.v0.0.1.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,267 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: clusterlogging.v0.0.1
+  namespace: placeholder
```
Comment (alecmerdler) [inline deploy/chart/catalog_resources/rh-operators/clusterlogging.v0.0.1.clusterserviceversion.yaml]: Can you add an example `ClusterLogging` JSON as an annotation, similar to the [ones for etcd Operator](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/deploy/chart/catalog_resources/rh-operators/etcdoperator.v0.9.2.clusterserviceversion.yaml#L10)?

File: deploy/chart/catalog_resources/rh-operators/elasticsearch.v0.0.1.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,174 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: elasticsearch.v0.0.1
+  namespace: placeholder
```
Comment (alecmerdler) [inline deploy/chart/catalog_resources/rh-operators/elasticsearch.v0.0.1.clusterserviceversion.yaml]: Can you add an example `ClusterLogging` JSON as an annotation, similar to the [ones for etcd Operator](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/deploy/chart/catalog_resources/rh-operators/etcdoperator.v0.9.2.clusterserviceversion.yaml#L10)?

File: deploy/chart/catalog_resources/rh-operators/elasticsearch.v0.0.1.clusterserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        - apiGroups:
+          - logging.openshift.io
+          resources:
+          - "*"
+          verbs:
+          - "*"
+        - apiGroups:
+          - ""
+          resources:
+          - pods
```
Comment (richm) [inline deploy/chart/catalog_resources/rh-operators/elasticsearch.v0.0.1.clusterserviceversion.yaml]: Need to add `pods/exec` here - see https://github.com/openshift/elasticsearch-operator/pull/41/files#diff-d9c51af46a881aab91ef7ababe2079dfR17

**Comment by richm** [comment]
> Holding until we can confirm both Operators work out of the box from an OLM Subscription.

How do we do that?  If I have a running cluster with olm, something like this?
- `oc edit -n  operator-lifecycle-manager cm/rh-operators`
- add the logging package definitions to `packages`
- add the logging crds to `customResourceDefinitions`
- add the logging csvs to `clusterServiceVersions`
?

File: deploy/chart/catalog_resources/rh-operators/elasticsearch.v0.0.1.clusterserviceversion.yaml (hunk 6 above)
Comment (jcantrill) [reply deploy/chart/catalog_resources/rh-operators/elasticsearch.v0.0.1.clusterserviceversion.yaml]: done.

**Comment by jcantrill** [comment]
> > Holding until we can confirm both Operators work out of the box from an OLM Subscription.
> 
> How do we do that? If I have a running cluster with olm, something like this?
> 
> * `oc edit -n  operator-lifecycle-manager cm/rh-operators`
> * add the logging package definitions to `packages`
> * add the logging crds to `customResourceDefinitions`
> * add the logging csvs to `clusterServiceVersions`
>   ?

Not sure if it is functional yet on the cluster but with @ecordell help I was able to deploy and display in the console:
```
$ oc scale -nopenshift-cluster-version --replicas=0 rs cluster-version-operator-7c5757d89
$ make release
...
wrote /tmp/tmp.u3y4Vk7V1f/chart/olm/templates/0000_30_06-rh-operators.configmap.yaml
...
$ oc replace --force -f /tmp/tmp.ivzg0towEf/chart/olm/templates/0000_30_06-rh-operators.configmap.yaml
```
Wait for a few minutes...
```
$ oc get packagemanifest
NAME                   AGE
svcat                  4h
couchbase-enterprise   4h
etcd                   4h
federationv2           4h
prometheus             4h
descheduler            8m
elasticsearch          8m
dynatrace-monitoring   4h
mongodb-enterprise     4h
amq-streams            4h
clusterLogging         8m
```

**Comment by jcantrill** [comment]
@alecmerdler @ecordell I have successfully deployed my operators though not logging (unrelated to OLM).  I'm satisfied this PR provides the opportunity to deploy cluster-logging.  It works assuming the image defined in the package manifest can be pulled which I assume will be resolved by the release process.   Only things I'm not sure if they are addressable by what changes we make here:
* elasticsearch-operator should not be visible in the UI;  there is no intention to allow users to deploy the EO without CLO
* Deleting the CLO subscription does not appear to remove the EO clusterserviceversion or affiliated service, deployment, replicaset, pod.  Should it?
* Not sure how the failed jobs relate to these changes.

**Comment by ecordell** [comment]
@jcantrill the e2e test is a flake (the installer failed to bring up the cluster). if you rebase, I would expect to see the unit test job pass, though

**Comment by jcantrill** [comment]
The most recent push combines the operators into a single deployment for now.  Rebased as well.  cc @alecmerdler

**Comment by ewolinetz** [comment]
https://github.com/openshift/cluster-logging-operator/pull/55 removes our operator's prior dependency on a signer cert (privileged operation) being provided to us.

**Comment by ecordell** [comment]
#617 merged and fixes `make schema-check` to actually be checking your manifests

**Pull request merged**